### PR TITLE
[SP-3214] Backport of MONDRIAN-2442 - Mondrian throws NullPointerExce…

### PR DESCRIPTION
**Pull request by stanislau-strelchanka**
[SP-3214] Backport of MONDRIAN-2442 - Mondrian throws NullPointerException with report that uses aggregate tables (6.1 Suite)

### CODE REVIEW

**Comment by stanislau-strelchanka** [comment]
@lucboudreau @mkambol @kurtwalker @mchen-len-son please review

**Comment by stanislau-strelchanka** [comment]
backport of #804

File: src/main/mondrian/olap/Mondrian.xml [hunk 1]
```diff
@@ -834,7 +834,7 @@ Revision is $Id$
             public Expression getNameExp() {
                 if (nameExp != null) {
                     return nameExp;
-                } else if (nameColumn != null) {
+                } else if (nameColumn != null &amp;&amp; !Util.equals(nameColumn, column)) {
```
Comment (lucboudreau) [inline src/main/mondrian/olap/Mondrian.xml]: That won't work. Copy paste error??

Comment (lucboudreau) [inline src/main/mondrian/olap/Mondrian.xml]: Scratch my previous comment. This is XML.

**Pull request merged**